src/layout/actions/focus.rs

```diff
@@ -32,10 +32,33 @@ impl LayoutTree {
         }
         let node_ix = self.tree.lookup_id(uuid)
             .ok_or(TreeError::NodeNotFound(uuid))?;
+        let parent_ix = self.tree.parent_of(node_ix)?;
+        match self.tree[parent_ix] {
+            Container::Container { layout, .. } => {
+                match layout {
+                    Layout::Tabbed | Layout::Stacked => {
+                        for child_ix in self.tree.children_of(parent_ix) {
```

Comment:
In this case you get children in random order. So eventually top window (not floating) will be the last one, but this is not the window you focus on. 
From my point of view, it's impossible to implement this logic w/o storing floating window pointer somewhere in the tree struct. In this case you could bring focused window in front and then floating.

Reply:
Actually this is possible, but floating window is something outstanding, so why not to store pointer it somewhere else

Reply:
It's not [actually random order](https://github.com/way-cooler/way-cooler/blob/master/src/layout/core/graph_tree.rs#L580) it's sorted by the focus weight, but yes for this purpose it might as well be.

If you look down at line 43, the only thing this loop is doing is sending the other windows to the back (i.e, just moving them behind the other tabbed/stacked that is already there). When you send all but the one you want to see to the back, the one you want to focus on is naturally at the front.

Is there an issue you can reproduce even with these changes? Or are you commenting purely on the code quality (which I admit is messy)

Reply:
I'm not sure if this is because of this change, but with floating window, there're a lot of strange behaviours. For example when the active container is not parent container of a floating window. If you change active views and later change floating window, behaviour is unpredictable. Sometimes static view becames just black.

Reply:
What if keep the code with `bring_to_front` of ficused view and then lookup for a floating window and make `bring_to_front` for it?

Reply:
That could work as well, feel free to make those changes and see if the floating issues you reported persist. Otherwise I can do it later tonight.